src/components/select/select.jsx

```diff
@@ -1,220 +1,71 @@
 import React, {
   forwardRef,
-  useEffect,
   useImperativeHandle,
   useRef,
   useState,
 } from 'react';
 import PropTypes from 'prop-types';
-import AbstractCustomField from '../__internal__/abstract-custom-field/abstract-custom-field.jsx';
-import Icon from '../icon/icon.jsx';
-import IconButton from '../icon-button/icon-button.jsx';
-import iconClear from '../../svgs/clear.svg';
-import iconCaretDown from '../../svgs/caret-down.svg';
-import iconCaretDownDisabled from '../../svgs/caret-down-disabled.svg';
-import styles from './select.css';
-import Listbox from '../listbox/listbox.jsx';
-import NoOptions from '../no-options/no-options.jsx';
-import useValidation from '../../hooks/use-validation.jsx';
-import useDropdownClose from '../../hooks/use-dropdown-close.js';
-import useMounted from '../../hooks/use-mounted.js';
+import FormControl from '../form-control/form-control.jsx';
+import SelectControl from '../control/select.jsx';
 import useForwardedRef from '../../hooks/use-forwarded-ref.js';
+import styles from './select.css';
 
-const Select = forwardRef(function Select(props, ref) {
-  const [showOptions, setShowOptions] = useState(false);
-  const [value, setValue] = useState(props.value === undefined ? null : props.value);
-  const [hasBeenBlurred, setBeenBlurred] = useState(false);
-  const [searchValue, setSearchValue] = useState(undefined);
-  const mounted = useMounted();
-  const listBoxRef = useRef();
-  const inputRef = useRef();
-  const selfRef = useForwardedRef(ref);
-
-  const [validationMessage, validate] = useValidation(props.errorText, inputRef);
-  const caretIcon = (<Icon
-    className={styles['input-icon']}
-    icon={props.readOnly ? iconCaretDownDisabled : iconCaretDown}
-    label={props.readOnly ? 'Select is not editable' : 'Open choices listbox'}
-  />);
-
-  const defaultValue = value ? props.displayValueEvaluator(value) : '';
+const Select = forwardRef(function Select(props, forwardedRef) {
+  const ref = useForwardedRef(forwardedRef);
+  const [invalid, setInvalid] = useState(false);
 
-  const wrapperRef = useRef(null);
-  const handleDropdownClose = () => {
-    setShowOptions(false);
-    setSearchValue(defaultValue);
+  const classNames = {
+    container: props.className,
   };
-  useDropdownClose(wrapperRef, showOptions, handleDropdownClose);
-
-  const clear = () => {
-    setValue(null);
-    setSearchValue(undefined);
-    inputRef.current.focus();
+  const ids = {
+    input: props.id,
+    label: `${props.id}-label`,
   };
-
-  const clearIcon = () => {
-    if (!props.readOnly && (value || searchValue)) {
-      return (<IconButton
-        className={styles['clear-button']}
-        icon={iconClear}
-        label={'Remove selected choice'}
-        onPress={clear}
-      />);
-    }
-
-    return undefined;
+  const refs = {
+    container: useRef(), // ?????
```

Comment:
what's the `???` about? Just the fact that a container uses a `ref`?

Reply:
Oh, it was my temporary TODO. It's like figure this out. Keep it or remove it. I _did_ figure it out but forgot to remove the comment.